src/main/java/io/rpg/model/actions/BaseAction.java

```diff
@@ -29,4 +34,26 @@ public void setEmitter(GameObject emitter) {
   public GameObject getEmitter() {
     return emitter;
   }
+
+  @Override
+  public void setAfterAction(Action action) {
+    afterAction = action;
+  }
+
+  @Override
+  public void setBeforeAction(Action action) {
+    beforeAction = action;
+  }
+
+  @Override
+  @Nullable
+  public Action getBeforeAction() {
```

Comment:
Maybe Optional<Action> would be better here?

Reply:
Good idea. Done here: [38c65fc](https://github.com/RPG-IO/rpg-io/pull/79/commits/38c65fce529153f5958672e365652700927e97ea)